x-team/xp-ui
# Implement lazy-load for Avatar img requests

**Release Type:** *Non-Breaking Feature*

Fixes https://x-team-internal.atlassian.net/browse/XP-3144

## Description

Implemented lazy-load for Avatar to load img files only when the Avatar is visible on viewport the first time

## Checklist


## Steps to Test or Reproduce

### XP-UI App
1. Run the app with the latest changes from this branch using the `npm run watch`

### XP App
1. Download the latest changes from `master`
2. Run the app linking the local `XP-UI` app following the fallback https://github.com/x-team/xp/wiki/Workflow-with-XP-and-XP-UI#fallback-solution
3. Go to http://localhost:8000/admin and open the webmaster tool to check the `network tab` and just filter by `img`
4. To see the lazyload effect start scrolling down the profile-list and meanwhile the avatars are in the viewport check how the image is downloaded.
5. Confirm that the images that aren't in the viewport don't have any image yet

## Impacted Areas in Application

Avatar UI component

## Screenshots

![2019-07-04 23 11 14](https://user-images.githubusercontent.com/6010411/60695818-32314300-9eb1-11e9-9369-d67f4a982b2a.gif)
